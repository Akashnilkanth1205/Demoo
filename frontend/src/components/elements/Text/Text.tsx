/**
 * @license
 * Copyright 2018-2020 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import "assets/css/write.scss"
import { IText } from "autogen/proto"
import classNames from "classnames"
import React, { ReactElement } from "react"

export interface TextProps {
  width: number
  element: IText
}

/**
 * Functional element representing preformatted (plain) text.
 */
export default function Text({ width, element }: TextProps): ReactElement {
  const styleProp = { width }

  return (
    <div className={classNames("fixed-width", "stText")} style={styleProp}>
      {element.body}
    </div>
  )
}
