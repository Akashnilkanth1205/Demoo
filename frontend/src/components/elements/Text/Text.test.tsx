/**
 * @license
 * Copyright 2018-2019 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from "react"
import { Map as ImmutableMap } from "immutable"
import Text from "./Text"
import renderer from "react-test-renderer"

const getProps = (elementProps: object = {}): Props => ({
  element: ImmutableMap({
    body: "some plain text",
    ...elementProps,
  }),
  width: 100,
})

describe("Text Element Test (preformatted text)", () => {
  it("renders plain text", () => {
    const props = getProps()
    const component = renderer.create(<Text {...props} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
