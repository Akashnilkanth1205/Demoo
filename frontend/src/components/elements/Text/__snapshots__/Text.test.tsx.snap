// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Element Test (preformatted text) renders plain text 1`] = `
<div
  className="fixed-width stText"
  style={
    Object {
      "width": 100,
    }
  }
>
  some plain text
</div>
`;
